test/unit/math/rev/arr/functor/map_rect_lpdf_mpi_test.cpp

```diff
@@ -0,0 +1,78 @@
+#include <stan/math.hpp>
+#include <stan/math/prim/arr/functor/mpi_command.hpp>
+#include <stan/math/prim/arr/functor/mpi_cluster.hpp>
+#include <stan/math/prim/arr/functor/map_rect_lpdf_mpi.hpp>
+#include <stan/math/rev/arr/functor/map_rect_lpdf_mpi.hpp>
+#include <boost/mpi.hpp>
+#include <boost/shared_ptr.hpp>
+
+#include <iostream>
+
+struct hard_work_lpdf {
+  template<typename T>
+  T operator()(const std::vector<T>& eta, const std::vector<T>& theta, const std::vector<double>& x_r, const std::vector<int>& x_i) const {
+    std::vector<T> res(2);
+    res[0] = theta[0]*theta[0];
+    res[1] = x_r[0]*theta[1]*theta[0];
+    T summed = log(res[0]*res[0] + res[1]*res[1]);
+    return(summed);
+  }
+
+  template<typename T>
+  static
+  T apply(const std::vector<T>& eta, const std::vector<T>& theta, const std::vector<double>& x_r, const std::vector<int>& x_i) {
+    const hard_work_lpdf f;
+    return f(eta, theta, x_r, x_i);
```

Comment:
Why this indirection?

Reply:
The indirection is not needed... I just thought that the Stan parser will generate a usual function object and then we would need this indirection.